addons/stock/models/product.py

```diff
@@ -375,6 +375,28 @@ def write(self, values):
                 raise UserError(msg)
         return res
 
+    def _compute_average_price(self, qty_done, quantity, moves):
```

Comment:
That could be defined on `stock.move`, since you loop on `moves` anyway (and you never use `self`). And that should probably be more logical to have it in `stock_account`.